bin/parse-legacy-file

```diff
@@ -3,18 +3,26 @@
 set -Eeuo pipefail
 
 get_legacy_version() {
-  local -r current_plugin="$1"
-  local -r version_file="$2"
-  local -r plugins_dir="${ASDF_DATA_DIR:-$HOME/.asdf}/plugins"
+  local -r version_file="$1"
   local -r supported_plugins=(terraform packer)
 
   # check if current plugin is supported and with a readable legacy file
   for plugin in "${supported_plugins[@]}"; do
-    if [[ ${current_plugin} =~ ${plugins_dir}/${plugin} &&
-      ${version_file} == *".${plugin}-version" && -r ${version_file} ]]; then
+    if [[ ${version_file} == *".${plugin}-version" && -r ${version_file} ]]; then
```

Comment:
I think this breaks if you have a `.terraform-version` and `.packer-version` in the same directory, since it will `cat` both of them instead of just the one the plugin is wanting to check. The `$0` passed to this function was used to deduce which legacy version file should be used because it has the plugin name in the path.